handler/src/main/java/io/netty/handler/ssl/OpenSslClientSessionCache.java

```diff
@@ -99,7 +99,7 @@ boolean setSession(long ssl, OpenSslSession session, String host, int port) {
                     if (toBeRemoved == null) {
                         toBeRemoved = new ArrayList<NativeSslSession>(2);
                     }
-                    toBeRemoved.add(nativeSslSession);
+                    toBeRemoved.add(sslSession);
```

Comment:
nit: is there a good way to test for this?

Reply:
The tests in https://github.com/netty/netty/pull/14358 catch it. I'll add a more focused test there.